experimental/kernels/kernels_c.h

```diff
@@ -0,0 +1,198 @@
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// --  USE_GPU_FOR_* are the GPU/CPU switching flags for the kernels in llm.c. --
+
+#define USE_GPU_FOR_ENCODER_FORWARD 1
```

Comment:
I get this can be convenient, but I wouldn't bake mixins in it at compile time like this. There may be reasons the type signatures will need to diverge and applications may intend to invoke a mixture of GPU / CPU implementations. 

My preference would be to just define the functions and leave it to the application / test layer to decide which they intend to or if / how they setup a dispatching mechanism. The kernel ibrary should just make the functions available without being responsible for comptime (or for that matter, any dispatch).

Am I missing anything, what do you think?

Reply:
When we are actually using these kernels in an application, it is better to put the layers of abstraction in there.
I am using the macros to see which GPU kernels are violating the test in llm.c.